chalice/config.py

```diff
@@ -79,13 +80,15 @@ class Config(object):
 
     def __init__(self,
                  chalice_stage=DEFAULT_STAGE_NAME,
+                 function_name=DEFAULT_HANDLER_NAME,
                  user_provided_params=None,
                  config_from_disk=None,
                  default_params=None):
-        # type: (str, StrMap, StrMap, StrMap) -> None
+        # type: (str, str, StrMap, StrMap, StrMap) -> None
         #: Params that a user provided explicitly,
         #: typically via the command line.
         self.chalice_stage = chalice_stage
+        self.function_name = function_name
```

Comment:
Ok cool. I was thinking along the same lines of how you would get resource specific configuration. Makes sense to keep the changes small for now until more configurable managed resources get added as the deployer will need to get refactored.